nanoparticle.py

```diff
@@ -24,17 +24,17 @@ class NanoParticle:
 	ligands = []
 	def __init__(
 		self,
-		x=0.0,
-		y=0.0,
-		z=0.0,
+#		x=0.0,
+#		y=0.0,
+#		z=0.0,
```

Comment:
Why did you get rid of this?

Reply:
I'll just cite myself from the commit message:
"I removed the (x,y,z) coordinates of the Nanoparticle because they are not it's immutable properties (so are not descriptive of a Nanoparticle, but rather it's state in a simulation), are never updated nor ever used anywhere (as far as I could tell)."

It's especially confusing because MembraneSimulation also has some coordinates, and also a rotation, and it makes sense that it does because those basically are parameters of the simulation and not of the particle, but then I wondered wtf are these Nanoparticle coordinates for... when I checked I saw they are not used anywhere and so thought it's OK to remove them, to reduce confusion in the code a bit...
There are more parameters like this that are a bit confusing, hardcoded around the place and/or never used or redundant (it seems to me), but I wanted to discuss it with you before I change anything further.

Reply:
These were originally intended to be the starting position of the nanoparticle in the simulation, not reflective of the simulation state once it starts running. Just a note, setting a value in a function call like this in Python is not the same as hardcoding, it defines a default value that allows the function to be called without requiring a value to be present, covering the most general use case.

You're right in this case, the values are not used, we didn't use them when we implemented the 3D model (presumably an oversight), np-specific coordinates were used in the 2D case I believe. It would be nice to reimplement this though, as simulations with more than one nanoparticle are part of our plans for the project.

Reply:
Yes, I know default values in Python are not hardcoding but I wasn't referring to these as hardcoding, sorry I expressed myself badly, I was just enumerating problematic traits of some stuff (I'll make an issue or something where I'll explain so we can discuss there), because there is some hardcoded stuff around, which would be better if it were defaults, but maybe can be gotten rid of completely.

In any case... when I first saw this I thought it weird that the MembraneSimulation has coordinates and the rotation, and thought that it should be moved from there to the Nanoparticle (in any case they shouldn't be in both places, and they should be together - the linear and the rotational coordinates I mean). But when I thought a bit more about it I realised what I said above, that actually the starting coordinates and the rotation of a nanoparticle (even if there are multiple) are actually parameters of the simulation, and not the particles themselves, in a sense.
So even if there would be more of them, that information could be stored in the MembraneSimulation, for example as a list of triplets (particle, location, rotation) or three lists or whatever...
You could make an argument to put it in the Nanoparticles themselves, but I don't think it makes sense to have rotations in one place and location coordinates in another... So this seemed an easier resolution at this point (and this dilemma is why I didn't delete them completely, just commented them out).
Hope that clears up everything :)